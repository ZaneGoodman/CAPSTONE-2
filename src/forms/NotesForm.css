.NotesForm-label {
  font-size: 18px;
  font-weight: 600;
}
.NotesForm-input {
  font-family: "Rosarivo", cursive;
}
