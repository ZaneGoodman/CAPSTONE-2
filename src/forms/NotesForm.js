import React, { useState, useRef } from "react";
import Notes from "../models/notes";
import "./NotesForm.css";
import Form from "react-bootstrap/Form";
import Button from "react-bootstrap/Button";
//** Form for adding notes for the current Rosary prayer. Will render underneath the Rosary */
const NotesForm = ({ date, season, username, setAddedNotes }) => {
  const checkbox = useRef();
  const INITIAL_STATE = {
    notes: "",
    marker: "false",
  };

  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleSubmit = async (evt) => {
    evt.preventDefault();
    await Notes.addNotes(
      username,
      formData.notes,
      formData.marker,
      date,
      season
    );
    /**Setting addedNotes to true on submit will tell the parent component to hide the form */
    setAddedNotes(true);
    setFormData(INITIAL_STATE);
  };

  const handleCheckbox = (evt) => {
    const { name } = evt.target;
    setFormData((fData) => ({
      ...fData,
      [name]: checkbox.current.checked,
    }));
  };

  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData((fData) => ({
      ...fData,
      [name]: value,
    }));
  };
  return (
    <>
      <Form>
        <Form.Group className="mb-3">
          <Form.Label className="NotesForm-label" htmlFor="notes">
            Notes
          </Form.Label>

          <Form.Control
            className="NotesForm-input"
            as="textarea"
            onChange={handleChange}
            id="notes"
            name="notes"
            rows={15}
            cols={100}
            value={formData.notes}
          />
        </Form.Group>
        <Form.Group className="mb-3">
          <Form.Label className="NotesForm-label" htmlFor="marker">
            Prayed today?
          </Form.Label>
          <Form.Check
            onChange={handleCheckbox}
            type="checkbox"
            id="marker"
            name="marker"
            ref={checkbox}
            value={formData.marker}
          />
        </Form.Group>
        <Button variant="secondary" onClick={handleSubmit}>
          Submit
        </Button>
      </Form>
    </>
  );
};

export default NotesForm;
