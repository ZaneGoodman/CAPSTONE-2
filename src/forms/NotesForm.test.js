import React from "react";
import { render } from "@testing-library/react";

import NotesForm from "./NotesForm";

//Smoke test
test("renders NotesForm component with authentication", async () => {
  render(
    <NotesForm
      date="2023/05/05"
      season="Ordinary Time"
      username="testuser"
      setAddedNotes={(boolean) => boolean}
    />
  );
});

//Snapshot test  <MemoryRouter>
it("NotesForm matches snapshot", async () => {
  const { asFragment } = render(
    <NotesForm
      date="2023/05/05"
      season="Ordinary Time"
      username="testuser"
      setAddedNotes={(boolean) => boolean}
    />
  );
  await expect(asFragment()).toMatchSnapshot();
});
