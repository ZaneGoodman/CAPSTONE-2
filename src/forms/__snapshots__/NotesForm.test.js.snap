// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesForm matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="notes"
    >
      Notes
    </label>
    <textarea
      cols="100"
      id="notes"
      name="notes"
      rows="15"
    />
    <label
      for="marker"
    >
      Prayed today?
    </label>
    <input
      id="marker"
      name="marker"
      type="checkbox"
      value="false"
    />
    <button>
      Submit
    </button>
  </form>
</DocumentFragment>
`;
